### ISSUE TITLE
Display the time...?

### ISSUE THREAD

**Issue by jasonm23** [Lambda-Emacs/lambda-line#2]
Any ideas why this isn't working?

```
(display-time-mode 1)
```

**Comment by mclearc** [comment]
Lambda line overwrites the normal mode line display and requires you to declare everything that shows. But this seems like an obvious thing that people would want to display so I’ll try and fix this soon. 


--
Colin McLear
Associate Professor
Department of Philosophy 
University of Nebraska–Lincoln
colinmclear.net

**Comment by jasonm23** [comment]
Thank you.  I can take a look at patching it if you are busy today.

**Comment by jasonm23** [comment]
I had a look today and I came up with this:

<img width="1065" alt="Screenshot 2022-08-31 at 12 00 39 PM" src="https://user-images.githubusercontent.com/71587/187596306-7c1adbfa-9015-4db9-b7a7-888150864219.png">

```lisp
(defun lambda-line-time ()
  (let* ((hour (string-to-number (format-time-string "%I")))
         (icon (all-the-icons-wicon (format "time-%s" hour) :height 1.3 :v-adjust 0.0)))
    (concat
     (propertize (format-time-string " %H:%M ") 'face `(:height 0.9))
     (propertize (format "%s " icon) 'face `(:height 1.0 :family ,(all-the-icons-wicon-family)) 'display '(raise -0.0)))))

(defun lambda-line-default-mode ()
  "Compose the default status line."
  (let ((buffer-name (format-mode-line (if buffer-file-name (file-name-nondirectory (buffer-file-name)) "%b")))
        (mode-name   (lambda-line-mode-name))
        (branch      (lambda-line-vc-project-branch))
        (position    (format-mode-line "%l:%c:%o")))
    (lambda-line-compose (lambda-line-status)
                        (lambda-line-truncate buffer-name lambda-line-truncate-value)
                        (concat "⋮ " mode-name
                                (when branch
                                  branch)
                                " ⋮"
                                (when display-time-mode (lambda-line-time)))
                        nil

                        ;; Narrowed buffer
                        (if (buffer-narrowed-p)
                            (concat
                             (propertize "⇥ "  'face `(:inherit lambda-line-inactive-secondary))
                             position " ")
                          position))
```

It's using all-the-icons for the animated time/hour icon.  Request for comments / feedback before I compose a PR.

**Comment by jasonm23** [comment]
It occurs to me that a few more customized variables would be handy.

- `lambda-line-modes-begin-indicator` default `"("`
- `lambda-line-modes-end-indicator` default `")"`

(I'm using the vertical ellipsis in my example)

**Comment by jasonm23** [comment]
#3

**Comment by mclearc** [comment]
Looks good. In the pic the vertical spacing seems a bit off. You might want to try tweaking the settings for line space and symbol position. Here's what I have in a :custom block for my use-package setup:

```emacs-lisp
  (lambda-line-space-top +.50)
  (lambda-line-space-bottom -.50)
  (lambda-line-symbol-position 0.1)
```
You can tweak these accordingly

**Comment by mclearc** [comment]
Closed via #3
